File: TP1/Q11/paRec.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

bool stop(char* s){
	return (strlen(s) && s[0]=='F' && s[1]=='I' && s[2]=='M');
}

bool palin(char* s, int i){
	if(i>=((strlen(s)/2)-1)) return (s[i]==s[strlen(s)-i-1]);
	else return (s[i]==s[strlen(s)-i-1] && palin(s, i+1));
}

bool pali(char* s){
	return palin(s, 0);
}

int main(){
	char* input = NULL;
	bool stopper = false;
	while(!stopper){
		input = malloc(1001 * sizeof(char));
		scanf(" %1000[^\n]", input);
		if(stop(input)) stopper = true;
		else if(pali(input)) printf("SIM\n");
		else printf("NAO\n");
		free(input);
	}
	return 0;
}

File: TP1/Q10/paRec.java
import java.util.Scanner;

public class paRec{
	public static boolean stop(String s){
		if(s.length()==3 && s.charAt(0)=='F' && s.charAt(1)=='I' && s.charAt(2)=='M') return true;
		else return false;
	}

	public static boolean pali(String s){
		return pali(s, 0);
	}

	public static boolean pali(String s, int i){
		if(i>=((s.length()/2)-1)) return (s.charAt(i)==s.charAt(s.length()-i-1));
		else return (s.charAt(i)==s.charAt(s.length()-i-1) && pali(s, i+1));
	}

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		String input = null;
		boolean stop = false;
		while(!stop){
			input = sc.nextLine();
			if(stop(input)) stop = true;
			else if(pali(input)) System.out.println("SIM");
			else System.out.println("NAO");
		}
	}
}
